lib/sail.c

```diff
@@ -164,13 +164,29 @@ void dec_str(sail_string *str, const mpz_t n)
 void hex_str(sail_string *str, const mpz_t n)
 {
   sail_free(*str);
-  gmp_asprintf(str, "0x%Zx", n);
```

Comment:
This one could be simplified to use %#Zx? And only do the negative/positive check for the upper case one?

Reply:
I just tested this and it drops the `0x` prefix when the argument is zero